src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ClientRestartSystem.java

```diff
@@ -0,0 +1,42 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+package org.terasology.ligthandshadow.componentsystem.controllers;
+
+import org.terasology.engine.entitySystem.entity.EntityRef;
+import org.terasology.engine.entitySystem.event.ReceiveEvent;
+import org.terasology.engine.entitySystem.systems.BaseComponentSystem;
+import org.terasology.engine.entitySystem.systems.RegisterMode;
+import org.terasology.engine.entitySystem.systems.RegisterSystem;
+import org.terasology.engine.logic.players.LocalPlayer;
+import org.terasology.engine.registry.In;
+import org.terasology.engine.rendering.nui.NUIManager;
+import org.terasology.engine.rendering.nui.layers.ingame.DeathScreen;
+import org.terasology.ligthandshadow.componentsystem.LASUtils;
+import org.terasology.ligthandshadow.componentsystem.events.ClientRestartEvent;
+import org.terasology.nui.layouts.miglayout.MigLayout;
+
+/**
+ * System to close game over screen once restart is complete.
+ */
+
+@RegisterSystem(RegisterMode.CLIENT)
+public class ClientRestartSystem extends BaseComponentSystem {
+    @In
+    LocalPlayer localPlayer;
+    @In
+    NUIManager nuiManager;
+
+    @ReceiveEvent
+    public void onClientRestart(ClientRestartEvent event, EntityRef clientEntity) {
+        if (localPlayer.getClientEntity().equals(clientEntity)) {
+            if (nuiManager.isOpen(LASUtils.DEATH_SCREEN)) {
+                DeathScreen deathScreen = (DeathScreen) nuiManager.getScreen(LASUtils.DEATH_SCREEN);
+                MigLayout migLayout = deathScreen.find("playerStatistics", MigLayout.class);
```

Comment:
I don't think this needs to be that specific here, so let's improve it while touching the code. The method we need is defined in `UILayout`, so we can use that type instead of requiring it to be a `MigLayout`.

https://github.com/MovingBlocks/TeraNUI/blob/b7503ba0d733cc409df9bedf7fa284f8656f5b4d/nui/src/main/java/org/terasology/nui/UILayout.java#L41